PowerShellGet/PSModule.psm1

```diff
@@ -11633,26 +12185,98 @@ function Get-InstalledModuleDetails
-                                                                   {
-                                                                       $InstalledModuleDetails
-                                                                   }
-                                                                }
-                                                                else
-                                                                {
-                                                                    if( (-not $MinimumVersion -or ($MinimumVersion -le $InstalledModuleDetails.PSGetItemInfo.Version)) -and 
-                                                                        (-not $MaximumVersion -or ($MaximumVersion -ge $InstalledModuleDetails.PSGetItemInfo.Version)))
-                                                                    {
-                                                                        $InstalledModuleDetails
-                                                                    }
-                                                                }
+
+                                                                Test-ItemPrereleaseVersionRequirements -InstalledItemDetails $InstalledModuleDetails `
+                                                                                                            -RequiredVersion $RequiredVersion `
+                                                                                                            -MinimumVersion $MinimumVersion `
+                                                                                                            -MaximumVersion $MaximumVersion
                                                             }
                                                         }
                                                     }
 }
 
+# Returns the installed item details object if it meets the Required, Minimum, and Maximum version bounds.
+function Test-ItemPrereleaseVersionRequirements
+{
+    [CmdletBinding()]
+    param(
+
+        [ValidateNotNullOrEmpty()]
+        [System.Collections.Hashtable]
+        $InstalledItemDetails,
+
+        [string]
+        $RequiredVersion,
+        
+        [string]
+        $MinimumVersion,
+
+        [string]
+        $MaximumVersion
+    )
+
+    $psgitemVersionPart,$psgitemPrereleasePart = $InstalledItemDetails.PSGetItemInfo.Version -split '-',2
+
+    if($RequiredVersion)
+    {
+        $requiredVersionPart,$requiredPrereleasePart = $RequiredVersion -split '-',2
+        
+        if (($requiredVersionPart -eq $psgitemVersionPart) -and 
+            ($requiredPrereleasePart -eq $psgitemPrereleasePart)) 
+        {
+            $InstalledItemDetails
+        }
+    }
+    else
+    {
+        $minimumBoundMet = $false
+        if ($MinimumVersion)
+        {
+            $minimumVersionPart,$minimumPrereleasePart = $MinimumVersion -split '-',2
+
+            if ($minimumVersionPart -le $psgitemVersionPart)

```

Comment:
>if ($minimumVersionPart -le $psgitemVersionPart [](start = 12, length = 47)

String comparison?
Please ensure that version comparisons are done with version type values instead of string values otherwise you will see unwanted comparison results. 
Please check for this string comparisons in other places too.